tests/testthat/test_utils.R

```diff
@@ -0,0 +1,303 @@
+
+})
+
+context('Velocity Calculation')
+test_that('Calculate velocity given acceleration and sampling rate',{
+  # actual function in utils: velocity  
+  expect_is(mhealthtools:::velocity(datAccel$x,100), 'numeric') # Check if output is in correct format
+  
+})
+
+test_that('Calculate and add Velocity column for the tidy sensor data',{
+  # actual function in utils: mutate_velocity 
+  testTibble <- dplyr::tibble(window = NA, error = "Error calculating velocity")  
+  
+  expect_is(mhealthtools:::mutate_velocity(datAccelTidy,100),'data.frame') # Check if output is in correct format
+  expect_equal(mhealthtools:::mutate_velocity(datAccel,100),testTibble) # Throw an error if input is not in correct format
+  
+})
+
+context('Displacement Calculation')
+test_that('Calculate Displacement given acceleration and sampling rate',{
+  # actual function in utils: displacement  
+  expect_is(mhealthtools:::displacement(datAccel$x,100), 'numeric') # Check if output is in correct format
+  
+})
+
+test_that('Calculate and add Displacement column for the tidy sensor data',{
+  # actual function in utils: mutate_displacement  
+  testTibble <- dplyr::tibble(window = NA, error = "Error calculating displacement")  
+  
+  expect_is(mhealthtools:::mutate_displacement(datAccelTidy,100),'data.frame') # Check if output is in correct format
+  expect_equal(mhealthtools:::mutate_displacement(datAccel,100),testTibble) # Throw an error if input is not in correct format
+  
+})
+
+context('ACF calculation')
+test_that('Construct a dataframe with ACF values given tidy sensor data',{
+  # actual function in utils: calculate_acf  
+  testTibble <- dplyr::tibble(window = NA, error = "Error calculating ACF")
+  
+  expect_is(mhealthtools:::calculate_acf(datAccelTidy),'data.frame') # Check if output is in correct format
+  expect_equal(mhealthtools:::calculate_acf(datAccel),testTibble) # Throw an error if input is not in correct format
+  
+})
+
+context('Tag and Identify outlier windows based on device rotation(using gravity)')
+test_that('Identify min, max gravity values for each window of tidy sensor data',{
+  # actual function in utils: tag_outlier_windows_  
+  
+  gravityVec <- dat$gravity$x 
+  expect_is(mhealthtools:::tag_outlier_windows_(gravityVec, 256, 0.5), 'data.frame')
+  # Check if output is in correct format. 256 Window length and 0.5 overlap
+  
+  gravityVec <- c(rep(NA,255),1) 
+  # All values are NA except for the last value, our window length is 256, same as the length of gravityVec
+  # min(gravityVec) and max(gravityVec) should be NA, so the testOutput needs to be a dataframe with one obs. for all
+  # the three output vars with max and min having NAs
+  
+  testOutput <- mhealthtools:::tag_outlier_windows_(seq(256), 256, 0.5) # Initialize the data
+  testOutput$max <- NA
+  testOutput$min <- NA
+  
+  expect_equal(mhealthtools:::tag_outlier_windows_(gravityVec, 256, 0.5), testOutput) 
+  # Min and Max should be NA if any window has NAs in it
```

Comment:
I explicitly inputted na.rm=TRUE when I wrote this function. Is there a good reason we want an entire window min/max to be marked as NA if a single value is NA?